kgl-math/src/commonMain/kotlin/com/kgl/math/Vectors.kt

```diff
@@ -0,0 +1,289 @@
+package com.kgl.math
+
+/**
+ * A generic two-component vector. Methods in this interface support only read-only access to the vector;
+ * read/write access is supported through the [MutableVector2] interface.
+ * @param T the type of component contained in the vector. The vector is invariant on its component type.
+ */
+interface Vector2<T> {
+
+	val x: T
+	val y: T
+
+	operator fun component1(): T = x
+	operator fun component2(): T = y
```

Comment:
My gut says these should be extension functions but I'm torn. What do you think?

Reply:
I'm not sure either. I kind of want to follow Jetbrains' lead from the primitive types in the stdlib

Reply:
implemented in commit https://github.com/Dominaezzz/kgl/pull/19/commits/a72d81516851cbe00d212f395542c2f29d0491e3